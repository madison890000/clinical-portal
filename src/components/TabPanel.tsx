import { PropsWithChildren } from 'react';

interface ITabPanel {
    value: string | number;
    index: string | number;
}

const TabPanel: React.FC<PropsWithChildren<ITabPanel>> = ({ children, value, index }) => {
    return (
        <div
            style={{
                display: value === index ? 'block' : 'none',
                backgroundColor: '#f6f8fa',
                padding: 12
            }}
        >
            {children}
        </div>
    );
};
export default TabPanel;
