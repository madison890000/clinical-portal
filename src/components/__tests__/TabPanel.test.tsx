import React from 'react';
import TabPanel from '../TabPanel';
import renderer from 'react-test-renderer';

test('renders TabPanel active correctly', () => {
    const TabPanelDom = renderer
        .create(
            <TabPanel value={1} index={1}>
                <div>Tab panel Content</div>
            </TabPanel>
        )
        .toJSON();
    expect(TabPanelDom).toMatchSnapshot();
});

test('renders TabPanel non-active correctly', () => {
    const TabPanelDom = renderer
        .create(
            <TabPanel value={0} index={1}>
                <div>Tab panel Content</div>
            </TabPanel>
        )
        .toJSON();
    expect(TabPanelDom).toMatchSnapshot();
});
