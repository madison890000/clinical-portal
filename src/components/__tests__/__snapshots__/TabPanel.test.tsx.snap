// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TabPanel active correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f6f8fa",
      "display": "block",
      "padding": 12,
    }
  }
>
  <div>
    Tab panel Content
  </div>
</div>
`;

exports[`renders TabPanel non-active correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f6f8fa",
      "display": "none",
      "padding": 12,
    }
  }
>
  <div>
    Tab panel Content
  </div>
</div>
`;
